src/instantiate.ts

```diff
@@ -188,26 +201,104 @@ export async function loadAllofExtension(context: vscode.ExtensionContext) {
 
       list.push(`Clear list`);
 
+      const listItems: vscode.QuickPickItem[] = list.map(item => ({ label: item }))
+
       const quickPick = vscode.window.createQuickPick();
-      quickPick.items = list.map(item => ({ label: item }));
+      quickPick.items = listItems
       quickPick.placeholder = `Enter file path (Format: LIB/SPF/NAME.ext or /home/xx/file.txt)`;
 
-      quickPick.onDidChangeValue(() => {
+      quickPick.onDidChangeValue(async () => {
         // INJECT user values into proposed values
         if (!list.includes(quickPick.value.toUpperCase())) quickPick.items = [quickPick.value.toUpperCase(), ...list].map(label => ({ label }));
+
+        if (quickPick.value.length >= 3) {
+          const selectionSplit = quickPick.value.split('/')
+          let resultSet: DB2Row[] = []
+          switch (selectionSplit.length) {
+            case 1:
+              resultSet = await content!.runSQL(`SELECT SYSTEM_SCHEMA_NAME, SCHEMA_TEXT FROM QSYS2.SYSSCHEMAS WHERE SYSTEM_SCHEMA_NAME NOT LIKE 'Q%' and SYSTEM_SCHEMA_NAME like upper('${quickPick.value}%') order by SYSTEM_SCHEMA_NAME `);
+
+              if (resultSet.length > 0) {
+                let listSchema: vscode.QuickPickItem[] = []
+                
+                resultSet.forEach(row => {
+                  listSchema.push({
+                    label: String(row.SYSTEM_SCHEMA_NAME),
+                    detail: String(row.SCHEMA_TEXT)
+                  })
+                })
+                
+                quickPick.items = [sourceSeparator,...listSchema,fileSeparator,...listItems]
+              }
+              
+              break;
+
+            case 2:
+              if (selectionSplit[1].length >= 3) {
+                resultSet = await content!.runSQL(`SELECT SYSTEM_TABLE_NAME, TEXT_DESCRIPTION FROM QSYS2.SYSFILES WHERE SYSTEM_TABLE_SCHEMA = '${selectionSplit[0]}' AND FILE_TYPE = 'SOURCE' and system_table_name like upper('${selectionSplit[1]}%') ORDER BY SYSTEM_TABLE_NAME`);
```

Comment:
Please use `QSYS2.SYSTABLES` instead of `SYSFILES`- the latter is a new contruct and may not be available on the server connected to by the user.

```suggestion
                resultSet = await content!.runSQL(`SELECT SYSTEM_TABLE_NAME, TEXT_DESCRIPTION FROM QSYS2.SYSTABLES WHERE SYSTEM_TABLE_SCHEMA = '${selectionSplit[0]}' AND FILE_TYPE = 'S
```